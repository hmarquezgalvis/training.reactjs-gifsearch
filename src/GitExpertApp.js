import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import GifGrid from './components/GifGrid';

export const GifExpertApp = ({ defaultState = []}) => {

  // const defaultState = [ 'One Punch', 'Dragon Ball', 'Demon Slayer' ];
  const [ categories, setCategories ] = useState(defaultState);

  return (
    <div className="container">
      <h2>GifExpertApp</h2>

      <AddCategory
        setCategories={ setCategories }
      />

      <hr/>

      { 
        categories.map( category => (
          <GifGrid 
            key={ category }
            category={ category }
          />
        ))
      }
    </div>
  );
};