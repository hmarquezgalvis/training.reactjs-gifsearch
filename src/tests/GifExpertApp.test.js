import { shallow } from "enzyme";
import { GifExpertApp } from "../GitExpertApp";

describe( 'tests for <GifExpertApp />', () => {

  test( 'must be right generated html', () => {
    const wrapper = shallow( <GifExpertApp /> );

    expect( wrapper ).toMatchSnapshot();
  });

  test('should show a list of categories', () => {
    const categories = [ 'One Punch', 'Dragon Ball', 'Demon Slayer' ];
    const wrapper = shallow( <GifExpertApp defaultState={ categories } /> );

    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find('GifGrid').length ).toBe( categories.length );
  })
  

});