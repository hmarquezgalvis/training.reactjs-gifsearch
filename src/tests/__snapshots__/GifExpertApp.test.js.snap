// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for <GifExpertApp /> must be right generated html 1`] = `
<div
  className="container"
>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</div>
`;

exports[`tests for <GifExpertApp /> should show a list of categories 1`] = `
<div
  className="container"
>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="One Punch"
    key="One Punch"
  />
  <GifGrid
    category="Dragon Ball"
    key="Dragon Ball"
  />
  <GifGrid
    category="Demon Slayer"
    key="Demon Slayer"
  />
</div>
`;
